### DISCUSSION TITLE
[Feature request] Add resolutions infos for each game page in the store

### DISCUSSION THREAD

**Discussion by Lordubuntu94** [ChristopherKlay/StadiaEnhanced#31] [Ideas & Requests]
For example, I go to the division 2 store page.
Near youtube and metacritic buttons you can add a label "Supported resolutions : 1080p 60fps / 4k 60fps"

For this task https://stadiahead.com/games might be handy

**Comment by Lordubuntu94** [comment]
Another thing that could be cool (but i'll write it there just to chat because i know it may be too hard to make) to have is a "Force 60fps button" into the "Stream" tab. With this feature the extension should be able to firstly know each game that runs at 60fps and then set the resolution once a game is launched. If there are both 1080 and 4k at 60fps it should prioritize the 4k unless 1080 is forced. If there is only 30fps then it does nothing. 
This one would be a kill feature for me. I hardly prefer framerate to stream quality.
It may be easier/possible to make once you will add resolutions profiles to each game and a dedicated list for game resolutions.

**Comment by ChristopherKlay** [comment]
The issue with both these great suggestions is that they rely on data that isn't actually available through Stadia, which creates a couple of issues.

- The data isn't actually verified in most cases. We've had cases where people thought the resolution of a game is 1440p (on 4K) for multiple weeks, just to get a confirmation that it's upscaled 1080p.
- The data isn't actually known for a lot of games.
- The data needs to be updated manually. Which means it's further delayed by updates to the extension (on top of the delay introduced by having to find out the resolution/fps in the first place).

In the end, a list of games and their resolution/fps (or better said "available data", since a lot is guessed/unknown) wouldn't be hard to add to Stadia Enhanced; Making fps filtering and such work consistently however definitely would be, because said list wouldn't be able to make sure that a lot of games (and all new ones) are actually 60fps at a specific resolution.

The suggestion is actually already on my to-do list, due to #25 which is effectively the same functionality. It just needs to wait until i can figure out a sustainable and reliable path for it.

**Reply by Manu99it** [reply]
What about [that airtable](https://airtable.com/shr32bmiOThVvSGar/tblAeJTnP2bzZyews) with many useful information about resolution and fps? It's much reliable and the maintainer (he's the review manager of StadiaSource.com, OriginalPenguin94 on reddit) try to get information confirmed by publishers or developers of games

**Reply by ChristopherKlay** [reply]
It would be possible to implement the infos from said table, but the process would be manually and thus new information wouldn't be available until the next patch comes in. On top of even being incorrect, if the devs change anything, like a added performance mode.

**Reply by Manu99it** [reply]
Wouldn't be possible to make a custom database to update that even without updating the extension but pushing an update to a dedicated server? Maybe some of already open databases for stadia could add those information to make it reusable by anyone

**Reply by Manu99it** [reply]
However I don't think is necessary in any case, even if it would be awesome I just like to play games 😅😅

**Reply by ChristopherKlay** [reply]
It would absolutely be possible.

While chrome extensions can't directly request such content (security restrictions based on the origin) it would be possible to make a request via the background page and fetch the data. Even injecting a function that requests the data into the actual Stadia page could work.

The issue here is mainly that this would require some kind of always up-to-date list that's easily reachable/readable and so far i limited Stadia Enhanced to work with what's provided via Stadias website, because doing so makes it not only very robust and easy to repair, it also keeps it easy to adapt - i.e. if Stadia would re-build their entire website, it wouldn't take a single week to get the extension up and running again.
